### ISSUE TITLE
sigh manage --clean_pattern works, -p doesnt

### ISSUE THREAD

**Issue by rompic** [fastlane/fastlane#8076]
### New Issue Checklist

- [X] Updated fastlane to the latest version
- [X] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
cleaning up files localy does work with --clean_pattern , but calling the same command with -p only lists profiles.

##### Complete output when running fastlane, including the stack trace and command used
fastlane sigh -v
fastlane installation at path:
/usr/local/lib/ruby/gems/2.3.0/gems/fastlane-2.13.0/bin/fastlane
-----------------------------
sigh 2.13.0

fastlane sigh manage --help

  manage

  Usage: fastlane sigh manage

  Manage installed provisioning profiles on your system.

  Examples:
        
    # Remove all "iOS Team Provisioning" provisioning profiles
    sigh manage -p "iOS\ ?Team Provisioning Profile"
        
  Options:
    -f, --force          Force remove all expired provisioning profiles. Required on CI. 
    -e, --clean_expired  Remove all expired provisioning profiles. 
    **-p, --clean_pattern STRING Remove any provisioning profiles that matches the regular** expression. 
    
fastlane sigh manage -p "AdHoc"
[17:08:16]: Loading Provisioning profiles from ~/Library/MobileDevice/Provisioning Profiles/
[17:08:21]: Provisioning profiles installed
[17:08:21]: Valid:
... (list of profiles)
[17:08:35]: 
[17:08:35]: Summary
[17:08:35]: 108 installed profiles
[17:08:35]: 0 are valid but will expire within 30 days
[17:08:35]: 108 are valid

fastlane sigh manage --clean_pattern "AdHoc"
[17:08:39]: Loading Provisioning profiles from ~/Library/MobileDevice/Provisioning Profiles/
[17:08:44]: The following provisioning profiles are either expired or matches your pattern:
Delete these provisioning profiles 0? (y/n)  n

**Comment by mgrebenets** [comment]
@hjanuschka, @KrauseFx, @asfalcone 

This seems to be related to https://github.com/fastlane/fastlane/issues/7935

The top level command short options will shadow subcommand short options.
Looks like this happens after monogem migration.

For example, `sigh resign -p` will be treated as `sigh -p`.

**Comment by hjanuschka** [comment]
@mgrebenets yeah - we should dive into this - and find a way how to deal with subcommands short options

**Comment by nafu** [comment]
@mgrebenets @hjanuschka 

> The top level command short options will shadow subcommand short options.

This behaivior can be confirmed with [commander](https://github.com/commander-rb/commander).
We can test with [runner_spec.rb](https://github.com/commander-rb/commander/blob/536ccfcf766151d6f7be890c467eaa1ac7fbed94/spec/runner_spec.rb#L83) like below.

```diff
diff --git a/spec/runner_spec.rb b/spec/runner_spec.rb
index ebff198..ff5dfca 100644
--- a/spec/runner_spec.rb
+++ b/spec/runner_spec.rb
@@ -263,6 +263,22 @@ describe Commander do
       expect(global_option).to eq('path')
       expect(command_option).to eq('bar')
     end
+
+    it 'should not disable local short options' do
+      global_option = nil
+      command_option = nil
+      new_command_runner('foo', '-p', 'bar') do
+        # Same at https://github.com/fastlane/fastlane/blob/42880c27c428b0875dc8dfbaba1ba14545f8a156/sigh/lib/sigh/commands_generator.rb#L25
+        global_option('-p', '--testing-command VALUE') { |v| global_option = v }
+
+        command :foo do |c|
+          # Same at https://github.com/fastlane/fastlane/blob/42880c27c428b0875dc8dfbaba1ba14545f8a156/sigh/lib/sigh/commands_generator.rb#L109
+          c.option('-p', '--duplicated-short-command VALUE') { |v| command_option = v }
+          c.when_called {}
+        end
+      end.run!
+
+      expect(global_option).to eq(nil) # Fail because it currently returns 'bar'
+      expect(command_option).to eq('bar') # Fail because it currently returns nil
+    end
   end
```

**Comment by hjanuschka** [comment]
thx for the testcase, this is a good starting point - now we need to figure out how to drop the parent short options - as i think its supposed that subcommands have their own context

**Comment by nafu** [comment]
OK.
If so, how about using global_option instead of c.option?
It will succeed the test case.

```diff
diff --git a/spec/runner_spec.rb b/spec/runner_spec.rb
index ebff198..ff5dfca 100644
--- a/spec/runner_spec.rb
+++ b/spec/runner_spec.rb
@@ -263,6 +263,22 @@ describe Commander do
       expect(global_option).to eq('path')
       expect(command_option).to eq('bar')
     end
+
+    it 'should not disable local short options' do
+      global_option = nil
+      command_option = nil
+      new_command_runner('foo', '-p', 'bar') do
+        # Same at https://github.com/fastlane/fastlane/blob/42880c27c428b0875dc8dfbaba1ba14545f8a156/sigh/lib/sigh/commands_generator.rb#L25
+        global_option('-p', '--testing-command VALUE') { |v| global_option = v }
+
+        command :foo do |c|
+          # Same at https://github.com/fastlane/fastlane/blob/42880c27c428b0875dc8dfbaba1ba14545f8a156/sigh/lib/sigh/commands_generator.rb#L109
+          global_option('-p', '--duplicated-short-command VALUE') { |v| command_option = v }
+          c.when_called {}
+        end
+      end.run!
+
+      expect(global_option).to eq(nil) # Success
+      expect(command_option).to eq('bar') # Success
+    end
   end
```

**Comment by mfurtak** [comment]
Hi @nafu 👋 

This is a very helpful investigation, thank you.

I believe this problem is a regression caused by introducing the `-p` `--platform` option to the _sigh_ tool options. I think the most direct solution is to remove the `-p` short flag from this new `--platform` option. /cc @hjanuschka

**Comment by nafu** [comment]
Hi @mfurtak 👋

That's true about `-p`. But `-i` is also affected.

short option | duplication occured SHA (PR)
--- | ---
`-p` | 790b6e89e334fe0162c3dbaa47fd5e2e78865926 (https://github.com/fastlane/fastlane/pull/6169)
`-i` | at least from 4da135cbe6c7e428fba308ad8c7c53b1c912a208 (https://github.com/fastlane/fastlane/pull/4579)

**Comment by mfurtak** [comment]
@nafu Yes, you're right!

I have proposed a fix in #8129 that should eliminate all such conflicts for _sigh_. (Other tools would need a similar fix, if we like the approach)

If you have the time, it would be very helpful if you test the solution in #8129 to see if you can find any regressions.

Thank you 👍

**Comment by KrauseFx** [comment]
Hey @nafu, can you drop me a quick email to contact@krausefx.com? 👍

**Comment by asfalcone** [comment]
I'm going to close out this issue because the fix was shipped in _fastlane_ 2.16.0 👍
